### ISSUE TITLE
Support for Unix ACLs?

### ISSUE THREAD

**Issue by danmosemsft** [dotnet/runtime#928]
What is our story for ACL's on Unix, where file system supports it?

[Mono.Posix](https://www.nuget.org/packages/Mono.Posix.NETStandard) (what we point at for API for Unix primitives) does not support them. Do we need to offer support for them somewhere? 

Whereever that support lies, do we need to add extension methods (like [this](https://github.com/dotnet/corefx/issues/41614)) to allow objects to be created with the ACL applied?  If not, what is the story for folks porting Windows code using ACL's to Linux?

This assumes we do not want to attempt to overload types like `DirectorySecurity`, which were designed with Windows in mind, to accommodate Unix.

**Comment by JeremyKuhne** [comment]
Triage: Conceptually we want to have ACL support for Linux (see [RedHat's support](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/storage_administration_guide/ch-acls)). The reason we haven't done this yet is that there isn't Posix support yet and there hasn't been significant asks from customers yet.

If we do it we'd also want to see if we can provide some sort of common abstraction with the existing Windows ACL support.

**Comment by jeffhandley** [comment]
I chatted with @carlossanlop on this and we decided to remove it from scope for 6.0. If we want to bring this into scope for a future release, we will need to convert this into a User Story and spend more time understanding the scenarios and identifying some projects that could utilize the feature.

**Comment by danmosemsft** [comment]
Agreed. I think when we are ready to visit this, we should probably step back and think about our story for Unix-specific API's all up, including what's on Mono.Posix.

**Comment by danmoseley** [comment]
@eerhardt  based on your recent doc should you close this as out of scope?
